### PR TITLE
[azure] support new marketpalce image 413.92.2023101700

### PR DESCRIPTION
**Pull request by jinyunma**
https://issues.redhat.com/browse/CORS-2725

1. update workflow `cucushift-installer-rehearse-azure-ipi-marketplace` to test image `413.92.2023101700` to provision all nodes on 4.14+.
2. update existing destructive  job to test NoPurchasePlan image to provision all nodes on 4.14+.
3. add post-check step `cucushift-installer-check-azure-osimage`

### CODE REVIEW

**Comment by gpei** [comment]
@jinyunma do we have a conclusion on if 4.13 should also use the new published "413.92.2023101700"? Or we just want to guarantee 4.14 is working fine currently

File: ci-operator/step-registry/ipi/conf/azure/osimage/ipi-conf-azure-osimage-commands.sh [hunk 1]
```diff
@@ -58,56 +58,64 @@ AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
...
 compute:
 - platform:
     azure:
       osImage:
-        publisher: redhat
+        publisher: imageInfo[0]
         offer: ${imageInfo[1]}
         sku: ${imageInfo[2]}
         version: ${imageInfo[3]}
+        ${os_image_plan}
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/osimage/ipi-conf-azure-osimage-commands.sh:110]: We need to set `os_image_plan` with an empty value in previous lines when `OS_IMAGE_PLAN` is empty?

And when `OS_IMAGE_PLAN` is set to `WithPurchasePlan`, we also assemble `os_image_plan`, right?

**Comment by jinyunma** [comment]
> do we have a conclusion on if 4.13 should also use the new published "413.92.2023101700"? 

not yet, didn't get response from EXDSP- 2164.

> Or we just want to guarantee 4.14 is working fine currently

yes, since manual testing is passed, so I plan to add it into CI, if there is any change for image version published in azure marketplace, I think we can just update ENV `OS_IMAGE_VERSION: 413.92.2023101700` on 4.14+ relevant jobs.

File: ci-operator/step-registry/ipi/conf/azure/osimage/ipi-conf-azure-osimage-commands.sh (hunk 1 above)
Comment (jinyunma) [reply ci-operator/step-registry/ipi/conf/azure/osimage/ipi-conf-azure-osimage-commands.sh:111]: yes, fixed.

File: ci-operator/step-registry/ipi/conf/azure/osimage/ipi-conf-azure-osimage-commands.sh [hunk 2]
```diff
@@ -58,56 +58,65 @@ AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
...
+# image plan is case-sensitive, make sure that publisher/offer/sku keep the same as plan
+if [[ "${OS_IMAGE_PLAN}" == "WithPurchasePlan" ]] || [[ "${OS_IMAGE_PLAN}" == "" ]]; then
+    imageInfo[0]=$(az vm image show --urn ${image} --query 'plan.publisher' -otsv)
+    imageInfo[1]=$(az vm image show --urn ${image} --query 'plan.product' -otsv)
+    imageInfo[2]=$(az vm image show --urn ${image} --query 'plan.name' -otsv)
+fi
+
+#set plan for os image, default value is WithPurchasePlan
+os_image_plan=""
+if [[ "${OS_IMAGE_PLAN}" != "" ]] && [[ ${ocp_minor_version} -ge 14 ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/osimage/ipi-conf-azure-osimage-commands.sh:96]: I guess we do not need to judge `ocp_minor_version` here, if someone set OS_IMAGE_PLAN on an unsupported version, that should be a job config issue.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml [hunk 3]
```diff
@@ -1740,12 +1739,24 @@ tests:
...
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
       E2E_RUN_TAGS: '@amd64 and @azure-ipi and @network-ovnkubernetes and not @fips'
-      OS_IMAGE_MASTERS: azureopenshift:aro4:413_v2:413.92.20230614
-      OS_IMAGE_MASTERS_PLAN: NoPurchasePlan
+      OS_IMAGE_VERSION: 413.92.2023101700
       TAG_VERSION: '@4.15'
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-azure-ipi-marketplace
+- as: azure-ipi-marketplace-noplan-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:1747]: How about merge the coverage into the above destructive ci? non-destructive cover `413.92.2023101700`, destructive one covers `azureopenshift:aro4:413_v2:413.92.20230614` ?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 4]
```diff
@@ -1835,25 +1835,23 @@ tests:
...
+- as: azure-ipi-marketplace-noplan-f28-destructive
+  cron: 24 12 16 * *
   steps:
     cluster_profile: azure-qe
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
       E2E_RUN_TAGS: '@amd64 and @azure-ipi and @network-ovnkubernetes and not @fips'
-      OS_IMAGE_MASTERS: azureopenshift:aro4:413_v2:413.92.20230614
-      OS_IMAGE_MASTERS_PLAN: NoPurchasePlan
+      OS_IMAGE_VERSION: 413.92.2023101700
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:1850]: We should cover `azureopenshift:aro4:413_v2:413.92.20230614` on this job, right?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml [hunk 5]
```diff
@@ -1727,21 +1727,21 @@ tests:
...
   steps:
     cluster_profile: azure-qe
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
       E2E_RUN_TAGS: '@amd64 and @azure-ipi and @network-ovnkubernetes and not @fips'
-      OS_IMAGE_MASTERS: azureopenshift:aro4:413_v2:413.92.20230614
-      OS_IMAGE_MASTERS_PLAN: NoPurchasePlan
+      OS_IMAGE_PLAN: NoPurchasePlan
+      OS_IMAGE_URN: azureopenshift:aro4:413_v2:413.92.20230614
+      OS_IMAGE_VERSION: 413.92.2023101700
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:1744]: From the codes, seem like we do not need `OS_IMAGE_VERSION` setting here?

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-marketplace-f28

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-marketplace-f28

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-marketplace-noplan-f28-destructive periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-ipi-workers-marketplace-f28

**Comment by jinyunma** [comment]
4.14-amd64-nightly-azure-ipi-marketplace-noplan-f28-destructive:  [job link](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/44693/rehearse-44693-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-marketplace-noplan-f28-destructive/1716651291331006464)

4.14-amd64-nightly-azure-ipi-marketplace-f28: [job link](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/44693/rehearse-44693-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-marketplace-f28/1716417646301286400)

4.13-amd64-nightly-azure-ipi-workers-marketplace-f28: [job link](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/44693/rehearse-44693-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-ipi-workers-marketplace-f28/1716651291398115328)

minor update: query image generation from `az vm image show` instead of image name.

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-marketplace-f28

**Comment by jinyunma** [comment]
/pj-rehearse ack

cc @liangxia to review

**Pull request merged**